### DISCUSSION TITLE
After upgrading to Storybook 7 I get no events under the actions tab

### DISCUSSION THREAD

**Discussion by Kwooda** [storybookjs/storybook#23345] [Help]
I am using the @storybook/web-components framework, trying to figure out how to get actions to appear in the actions tab. I have followed everything I could find in the docs, but nothing works. The only time I've ever seen an action appear in Storybook 7 (under a react framework), was if the story did not define a render method.

I've tried setting argTypes in the stories file.
```
const argTypes = {
	onClick: { action: 'clicked' },
}
```
I've tried setting parameters in preview.js.
```
export default {
	parameters: {
		actions: { argTypesRegex: '^on.*' }
	}
  };
```

I originally had stories written in Storybook 6.5 and they worked fine, and I did a conversion to bring those stories into Storybook 7, and not a single one of them shows up in the actions.

How do I get this to work?

**Comment by lentilz** [comment]
I've noticed that on a fresh install of v7, @storybook/addon-actions isnt listed as a dev dependency either. However when looking at the actions readme https://github.com/storybookjs/storybook/tree/next/code/addons/actions it makes it seem like it should be explicitly installed and added to the main.js add-ons vs assuming its bundled into the essentials add on.

I wonder if this is related to why the tab is there but doesnt log the events

**Reply by lentilz** [reply]
After downgrading from v7.2 and following the approach in the workaround [from this issue](https://github.com/storybookjs/storybook/issues/14933#issuecomment-920578274), i can finally see a working version of something logging in the Actions tab however, the event name displays as a control in the argsTable which isnt ideal. Hiding it with the exclude parameter returns a unknown function for that event.

The action data retrieved then only shows the details from the Window event (location, CONFIG_TYPE: "DEVELOPMENT", STORIES array  etc) which isnt useful, rather than details from the component event itself, which isnt really useful

**Reply by mbulfair** [reply]
Not sure if you resolved it, but I did with `import { withActions } from '@storybook/addon-actions/decorator';`  in my preview.ts and adding 
```ts
 decorators: [
    withActions,
```
